src/components/QuestionnaireForm/QuestionnaireForm.jsx

```diff
@@ -55,16 +54,72 @@ export default class QuestionnaireForm extends Component {
     }
 
     const items = this.props.qform.item;
-    this.prepopulate(items, newResponse.item, saved_response)
+    this.prepopulate(items, newResponse.item, false)
+    this.state.savedResponse = newResponse;
+  }
+
+  componentDidMount() {
+    this.loadAndMergeForms(this.state.savedResponse);
+  }
+
+  loadPreviousForm() {
```

Comment:
this also fails without indication, the user should be notified if there's an error getting the responses.

Reply:
Fixed